ufl/domain.py

```diff
@@ -126,6 +153,119 @@ def _ufl_sort_key_(self):
         typespecific = (self._ufl_id, self._ufl_coordinate_element)
         return (self.geometric_dimension(), self.topological_dimension(), "Mesh", typespecific)
 
+    @property
+    def meshes(self):
+        """Return the component meshes."""
+        return (self,)
+
+    def iterable_like(self, element):
```

Comment:
Typing-hinting for new functions in UFL would be nice.

Reply:
This should be type-hinted as it has assumptions on the fact that element has the property `sub_elements`.

Reply:
Why not use `num_sub_elments` as in `mixedmesh`?

Reply:
Added type-hinting. Right, we now use `num_sub_elements`.